### PR TITLE
Fixes #11478 trigger tab when emmet expansion is the same

### LINKED ISSUE #11478: Can't insert a tab at the end of a line in a jade file
Have a jade file with this content-

```
extends layout

block content
  h1= title
  p Welcome to #{title}
```

Put the cursor directly after layout, content, or title, and press tab, and nothing happens. If you type a space, then you can insert a tab.

If the cursor is after the } in the last line, pressing tab will replace the braces with spaces and move the cursor next to the #.

### PR DESCRIPTION
**Pull request by ramya-rao-a**
cc @mrmlnc 

When expanded value from emmet is the same as the text it intends to replace, then treat it as no expansion has occured and trigger tab

### CODE REVIEW

File: src/vs/workbench/parts/emmet/node/editorAccessor.ts [hunk 1]
```diff
@@ -115,6 +116,12 @@ export class EditorAccessor implements emmet.Editor {
 		}
 
 		let range = new Range(startPosition.lineNumber, startPosition.column, endPosition.lineNumber, endPosition.column);
+		let textToReplace = this._editor.getModel().getValueInRange(range);
+		if (this._emmetActionName === 'expand_abbreviation' && (value === textToReplace || value === textToReplace + '${0}')) {
```
Comment (mrmlnc) [inline src/vs/workbench/parts/emmet/node/editorAccessor.ts]: Maybe divide into two variables?

```ts
const isExpandAbbreviation = this._emmetActionName === 'expand_abbreviation';
const isSameExpansion = value === textToReplace || value === textToReplace + '${0}';
```

File: src/vs/workbench/parts/emmet/node/emmetActions.ts [hunk 2]
```diff
@@ -261,17 +273,15 @@ export abstract class EmmetEditorAction extends EditorAction {
...
-
 	constructor(id: string, label: string, alias: string, actionName: string, kbOpts?: ICommandKeybindingsOptions) {
 		super({
 			id: id,
 			label: label,
 			alias: alias,
 			precondition: EditorContextKeys.Writable,
-			kbOpts: kbOpts
+			kbOpts: kbOpts,
+			actionName: actionName
```
Comment (mrmlnc) [inline src/vs/workbench/parts/emmet/node/emmetActions.ts]: We can use here ES2015 object destructuring for `id`, `label`, `alias`, `kbOpts`, `actionName`?

**Review by mrmlnc — APPROVED** [review]
I think it should work. Please add some tests for this case to `vscode/src/vs/workbench/parts/emmet/test/node/editorAccessor.test.ts` 👍

**Comment by ramya-rao-a** [comment]
Updated to add test case for replace range and telemetry for all emmet actions

**Review by egamma — APPROVED** [review]
👍  @ramya-rao-a assigning the PR back to you for the merge.

**Comment by ramya-rao-a** [comment]
Thanks @egamma and @mrmlnc

**Pull request merged**